tools/resource-gen/genutils/util.go

```diff
@@ -95,17 +98,17 @@ func ToResourceInfo(desc protoreflect.MessageDescriptor) ResourceInfo {
 			}
 		}
 	}
+	for i, c := range out.ResourceType {
+		if unicode.IsUpper(c) && i != 0 {
+			out.SingularDisplayName += " "
+		}
+		out.SingularDisplayName += string(c)
+	}
 	// Working around the fact we don't really differentiate policies from the rest of resources:
 	// Anything global can't be a policy as it need to be on a mesh. Anything with locked Ws config is something internal and therefore not a policy
-	out.IsPolicy = !out.SkipRegistration && !out.Global && !out.WsAdminOnly && !out.WsReadOnly && out.ResourceType != "Dataplane"
-	if out.DisplayName == "" && out.IsPolicy {
-		for i, c := range out.ResourceType {
-			if unicode.IsUpper(c) && i != 0 {
-				out.DisplayName += " "
-			}
-			out.DisplayName += string(c)
-		}
-		out.DisplayName += "s"
+	out.IsPolicy = !out.SkipRegistration && !out.Global && !out.WsAdminOnly && !out.WsReadOnly && out.ResourceType != "Dataplane" && out.ResourceType != "ExternalService"
+	if out.PluralDisplayName == "" && out.IsPolicy {
+		out.PluralDisplayName = out.SingularDisplayName + "s"
```

Comment:
There is more complicated code to make a plural name https://github.com/kumahq/kuma/blob/master/tools/policy-gen/protoc-gen-kumapolicy/utils.go#L33. Maybe it should be reused.

Reply:
done